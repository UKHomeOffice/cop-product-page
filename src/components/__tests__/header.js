import React from "react";
import renderer, { create } from "react-test-renderer";
import Header from "../header";

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Header />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("menu button renders correctly when active", () => {
    const tree = create(<Header />);
    const instance = tree.root;

    const mButton = global.document.createElement('button');
    mButton.setAttribute('id', 'menuButton');
    mButton.setAttribute('classList', 'govuk-header__navigation--open');
    global.window.domNode = mButton;
    global.document.body.appendChild(mButton);

    const mNavigation = global.document.createElement('ul');
    mNavigation.setAttribute('id', 'navigation');
    mNavigation.setAttribute('classList', 'govuk-header__navigation--open');
    global.window.domNode = mNavigation;
    global.document.body.appendChild(mNavigation);
    const button = instance.findByProps({ id: 'menuButton' });
    button.props.onClick();
    expect(tree).toMatchSnapshot();

  });

  it("menu button renders correctly when inactive", () => {
    const tree = create(<Header />);
    const instance = tree.root;

    const mButton = global.document.createElement('button');
    mButton.setAttribute('id', 'menuButton');
    mButton.removeAttribute('classList', 'govuk-header__navigation--open');
    global.window.domNode = mButton;
    global.document.body.appendChild(mButton);

    const mNavigation = global.document.createElement('ul');
    mNavigation.setAttribute('id', 'navigation');
    mNavigation.removeAttribute('classList', 'govuk-header__navigation--open');
    global.window.domNode = mNavigation;
    global.document.body.appendChild(mNavigation);
    const button = instance.findByProps({ id: 'menuButton' });
    button.props.onClick();
    expect(tree).toMatchSnapshot();

  });

  it("displays the logo with correct source", () => {
    const tree = renderer
      .create(<Header />);
    const testInstance = tree.root;
    expect(testInstance.findByProps({ id: "logo-image" })._fiber.memoizedProps.source).toEqual("/images/logo.png");
  });
});
