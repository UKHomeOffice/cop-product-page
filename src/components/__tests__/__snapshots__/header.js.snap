// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header menu button renders correctly when active 1`] = `
<header
  className="govuk-header "
  data-module="header"
  role="banner"
>
  <div
    className="govuk-header__container govuk-width-container"
  >
    <div
      className="govuk-header__logo"
    >
      <a
        className="govuk-header__link govuk-header__link--homepage"
        href="/"
      >
        <span
          style={
            Object {
              "marginRight": "6px",
            }
          }
        >
          <img
            alt="logo"
            className="cop-logo"
            src="/images/logo.png"
          />
        </span>
        <span
          className="govuk-header__product-name service-name"
          id="service-name"
        />
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
      </a>
    </div>
    <div
      className="govuk-header__content"
    >
      <button
        aria-controls="navigation"
        aria-label="Show or hide Top Level Navigation"
        className="govuk-header__menu-button js-header-toggle"
        id="menuButton"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        type="button"
      >
        Menu
      </button>
      <nav>
        <ul
          aria-label="Top Level Navigation"
          className="govuk-header__navigation govuk-header__navigation--end"
          id="navigation"
        >
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="/about/"
            >
              About
            </a>
          </li>
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="/help/"
            >
              Help
            </a>
          </li>
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="https://www.cop.homeoffice.gov.uk/"
              rel="noreferrer"
              target="_blank"
            >
              Sign-in
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header menu button renders correctly when inactive 1`] = `
<header
  className="govuk-header "
  data-module="header"
  role="banner"
>
  <div
    className="govuk-header__container govuk-width-container"
  >
    <div
      className="govuk-header__logo"
    >
      <a
        className="govuk-header__link govuk-header__link--homepage"
        href="/"
      >
        <span
          style={
            Object {
              "marginRight": "6px",
            }
          }
        >
          <img
            alt="logo"
            className="cop-logo"
            src="/images/logo.png"
          />
        </span>
        <span
          className="govuk-header__product-name service-name"
          id="service-name"
        />
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
      </a>
    </div>
    <div
      className="govuk-header__content"
    >
      <button
        aria-controls="navigation"
        aria-label="Show or hide Top Level Navigation"
        className="govuk-header__menu-button js-header-toggle"
        id="menuButton"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        type="button"
      >
        Menu
      </button>
      <nav>
        <ul
          aria-label="Top Level Navigation"
          className="govuk-header__navigation govuk-header__navigation--end"
          id="navigation"
        >
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="/about/"
            >
              About
            </a>
          </li>
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="/help/"
            >
              Help
            </a>
          </li>
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="https://www.cop.homeoffice.gov.uk/"
              rel="noreferrer"
              target="_blank"
            >
              Sign-in
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header renders correctly 1`] = `
<header
  className="govuk-header "
  data-module="header"
  role="banner"
>
  <div
    className="govuk-header__container govuk-width-container"
  >
    <div
      className="govuk-header__logo"
    >
      <a
        className="govuk-header__link govuk-header__link--homepage"
        href="/"
      >
        <span
          style={
            Object {
              "marginRight": "6px",
            }
          }
        >
          <img
            alt="logo"
            className="cop-logo"
            src="/images/logo.png"
          />
        </span>
        <span
          className="govuk-header__product-name service-name"
          id="service-name"
        />
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
      </a>
    </div>
    <div
      className="govuk-header__content"
    >
      <button
        aria-controls="navigation"
        aria-label="Show or hide Top Level Navigation"
        className="govuk-header__menu-button js-header-toggle"
        id="menuButton"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        type="button"
      >
        Menu
      </button>
      <nav>
        <ul
          aria-label="Top Level Navigation"
          className="govuk-header__navigation govuk-header__navigation--end"
          id="navigation"
        >
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="/about/"
            >
              About
            </a>
          </li>
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="/help/"
            >
              Help
            </a>
          </li>
          <li
            className="govuk-header__navigation-item"
          >
            <a
              className="govuk-header__link header-nav-items"
              href="https://www.cop.homeoffice.gov.uk/"
              rel="noreferrer"
              target="_blank"
            >
              Sign-in
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
